cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/ExpressionHandler.kt

```diff
@@ -470,8 +470,13 @@ class ExpressionHandler(lang: JavaLanguageFrontend) :
 
     private fun handleThisExpression(expr: Expression): Reference {
         val thisExpr = expr.asThisExpr()
-        val resolvedValueDeclaration = thisExpr.resolve()
-        val type = this.objectType(resolvedValueDeclaration.qualifiedName)
+        var qualifiedName: String
+        try {
+            qualifiedName = thisExpr.resolve().qualifiedName
+        } catch (use: UnsolvedSymbolException) {
+            qualifiedName = frontend.scopeManager.currentRecord?.name.toString()
```

Comment:
Can't we just use our scope manager function in all cases?

Reply:
At least our tests work when we always use our scope manager.

Reply:
ok we kann do that